src/pathfinding_service/api.py

```diff
@@ -122,6 +122,18 @@ def post(self, token_network_address: str) -> Tuple[dict, int]:
                 **optional_args,
             )
         except (NetworkXNoPath, NodeNotFound):
+
+            ## this is for assertion via the scenario player
+            if len(g.last_request) > 20: g.last_request.pop(0)
```

Comment:
You might want to use `collections.deque` with `maxlen=20`: https://docs.python.org/2/library/collections.html#collections.deque

Reply:
thanks, that is a nice object